src/Plugin/CookiePlugin.php

```diff
@@ -69,7 +69,11 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
                     }
 
                     // Restrict setting cookie from another domain
-                    if (false === strpos($cookie->getDomain(), $request->getUri()->getHost())) {
+                    if (false === strpos(
+                            '.'.$request->getUri()->getHost(),
+                            '.'.$cookie->getDomain()
```

Comment:
Isn't this check too weak (see https://3v4l.org/nXCMW)? You should here test that `$request->getUri()->getHost()` ends with `$cookie->getDomain()`.